lib/handler/status/durations.c

```diff
@@ -125,12 +136,19 @@ static h2o_iovec_t durations_status_final(void *priv, h2o_globalconf_t *gconf, h
             "request-total-time") "," DURATION_FMT("process-time") "," DURATION_FMT("response-time") "," DURATION_FMT("duration"),
         DURATION_VALS(connect_time), DURATION_VALS(header_time), DURATION_VALS(body_time), DURATION_VALS(request_total_time),
         DURATION_VALS(process_time), DURATION_VALS(response_time), DURATION_VALS(total_time));
-
-#undef BUFSIZE
 #undef DURATION_FMT
 #undef DURATION_VALS
-
+    char *delim = "";
+    ret.len += sprintf(ret.base + ret.len, ",\n\"evloop-latency-nanosec\": [");
+    for(int i = 0; i < agg_stats->stats.evloop_latency_nanosec.size; i++) {
+        ret.len += snprintf(ret.base + ret.len, BUFSIZE - ret.len, "%s%llu",
+                            delim, agg_stats->stats.evloop_latency_nanosec.entries[i]);
```

Comment:
I think you need to expand the buffer size (or at least check that we are not overrunning) now that we have a loop depending on a user-configured variable (i.e. `num-threads`).

Reply:
Thank you for this critical feedback. I added a check before writing to the buffer in 967c46e7298f3c7e5b1b55ea672b460bdbfddf45.